### PR TITLE
Adding 7.29.0-rc.1 to the release.json

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?
Adding 7.29.0-rc.1 to the release.json

### CODE REVIEW

**Review by KSerrania — COMMENTED** [review]
Could you rebase on the latest master? I see you're a few commits behind, and the `WINDOWS_DDNPM_VERSION` you're using (`0.98.2.git.74.86e7c5b`) doesn't match the one used on `master`'s nightly entries: https://github.com/DataDog/datadog-agent/blob/b30940ca84d750bb5f137c591e24d4d5e8154e77/release.json#L10 (`0.98.2.git.86.53d1ee4`).

**Review by olivielpeau — APPROVED** [review]

**Comment by olivielpeau** [comment]
This supersedes https://github.com/DataDog/datadog-agent/pull/8281, and let's fix the conflict on the `go.mod` file

**Review by olivielpeau — APPROVED** [review]

File: release.json [hunk 1]
```diff
@@ -22,6 +22,30 @@
...
+        "INTEGRATIONS_CORE_VERSION": "7.29.0-rc.1",
+        "OMNIBUS_SOFTWARE_VERSION": "7.29.0-rc.1",
+        "OMNIBUS_RUBY_VERSION": "7.29.0-rc.1",
+        "JMXFETCH_VERSION": "0.44.0",
+        "JMXFETCH_HASH": "758717a2fd9c50247336318644e9c985d9a8a9845039d59a8793923b3f76f494",
+        "SECURITY_AGENT_POLICIES_VERSION": "v0.10",
+        "MACOS_BUILD_VERSION": "6.29.0-rc.1",
+        "WINDOWS_DDNPM_DRIVER": "release-signed",
+        "WINDOWS_DDNPM_VERSION": "0.98.2.git.74.86e7c5b",
+        "WINDOWS_DDNPM_SHASUM": "61e13fccdca7bf6a356e7ef552a7845316b9905b5eee264811982e3034a3bb84"
```
Comment (KSerrania) [inline release.json:35]: Could you update these lines with what's used in `nightly`?
```suggestion
        "WINDOWS_DDNPM_VERSION": "0.98.2.git.86.53d1ee4",
        "WINDOWS_DDNPM_SHASUM": "5d31cbf7aea921edd5ba34baf074e496749265a80468b65a034d3796558a909e"
```

File: release.json [hunk 2]
```diff
@@ -22,6 +22,30 @@
...
+        "INTEGRATIONS_CORE_VERSION": "7.29.0-rc.1",
+        "OMNIBUS_SOFTWARE_VERSION": "7.29.0-rc.1",
+        "OMNIBUS_RUBY_VERSION": "7.29.0-rc.1",
+        "JMXFETCH_VERSION": "0.44.0",
+        "JMXFETCH_HASH": "758717a2fd9c50247336318644e9c985d9a8a9845039d59a8793923b3f76f494",
+        "SECURITY_AGENT_POLICIES_VERSION": "v0.10",
+        "MACOS_BUILD_VERSION": "7.29.0-rc.1",
+        "WINDOWS_DDNPM_DRIVER": "release-signed",
+        "WINDOWS_DDNPM_VERSION": "0.98.2.git.74.86e7c5b",
+        "WINDOWS_DDNPM_SHASUM": "61e13fccdca7bf6a356e7ef552a7845316b9905b5eee264811982e3034a3bb84"
```
Comment (KSerrania) [inline release.json:47]: Could you update these lines with what's used in `nightly-a7`?
```suggestion
        "WINDOWS_DDNPM_VERSION": "0.98.2.git.86.53d1ee4",
        "WINDOWS_DDNPM_SHASUM": "5d31cbf7aea921edd5ba34baf074e496749265a80468b65a034d3796558a909e"
```

**Pull request merged**